src/js/components/Accordion/Accordion.js

```diff
@@ -7,85 +7,76 @@ import { AccordionContext } from './AccordionContext';
 const activeAsArray = active =>
   typeof active === 'number' ? [active] : active;
 
-class Accordion extends Component {
-  static defaultProps = {
-    animate: true,
-    messages: {
-      tabContents: 'Tab Contents',
-    },
-  };
-
-  static getDerivedStateFromProps(nextProps, prevState) {
-    const { activeIndex } = nextProps;
-    const {
-      activeIndexes: stateActiveIndexes,
-      activeIndex: stateActiveIndex,
-    } = prevState;
-
-    const activeIndexes = activeAsArray(activeIndex) || [];
-
-    if (
-      (typeof activeIndex !== 'undefined' ||
-        activeIndex !== stateActiveIndex) &&
-      activeIndexes.join() !== stateActiveIndexes.join()
-    ) {
-      return { activeIndexes, activeIndex };
-    }
-
-    return null;
+function Accordion({
+  onActive,
+  multiple,
+  animate,
+  children,
+  messages,
+  activeIndex,
+  ...restProps
+}) {
+  const [activeIndexes, setActiveIndexes] = useState([]);
+  const [stateActiveIndex, setStateActiveIndex] = useState();
+
+  // Derived state from props
+  // https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops
+  const derivedActiveIndexes = activeAsArray(activeIndex) || [];
+  if (
+    (typeof activeIndex !== 'undefined' || activeIndex !== stateActiveIndex) &&
+    derivedActiveIndexes.join() !== activeIndexes.join()
+  ) {
+    setActiveIndexes(derivedActiveIndexes);
+    setStateActiveIndex(activeIndex);
   }
 
-  state = {
-    activeIndexes: [],
-  };
-
-  onPanelChange = index => {
-    const { activeIndexes } = this.state;
+  const onPanelChange = index => {
     let nextActiveIndexes = [...(activeIndexes || [])];
-    const { onActive, multiple } = this.props;
 
-    const activeIndex = nextActiveIndexes.indexOf(index);
-    if (activeIndex > -1) {
-      nextActiveIndexes.splice(activeIndex, 1);
+    const nextActiveIndex = nextActiveIndexes.indexOf(index);
+    if (nextActiveIndex > -1) {
+      nextActiveIndexes.splice(nextActiveIndex, 1);
     } else if (multiple) {
       nextActiveIndexes.push(index);
     } else {
       nextActiveIndexes = [index];
     }
 
-    this.setState({ activeIndexes: nextActiveIndexes }, () => {
-      if (onActive) {
-        onActive(nextActiveIndexes);
-      }
-    });
+    setActiveIndexes(nextActiveIndexes);
+    if (onActive) {
+      onActive(nextActiveIndexes);
+    }
   };
 
-  render() {
-    const { animate, children, messages, ...rest } = this.props;
-    const { activeIndexes } = this.state;
-
-    delete rest.onActive;
-
-    return (
-      <Box role="tablist" {...rest}>
-        {Children.toArray(children).map((panel, index) => (
-          <AccordionContext.Provider
-            key={`accordion-panel_${index + 0}`}
```

Comment:
One more question here, does `index + 0` won't make any difference I believe? Should I remove that?

Reply:
Oh I get it now. It is there to silence `eslint(react/no-array-index-key)` error message.